components/board/i2c.go

```diff
@@ -41,3 +43,33 @@ func (reg *I2CRegister) ReadByteData(ctx context.Context) (byte, error) {
 func (reg *I2CRegister) WriteByteData(ctx context.Context, data byte) error {
 	return reg.Handle.WriteByteData(ctx, reg.Register, data)
 }
+
+// ReadByteDataFromBus opens a handle for the bus adhoc to perform a single read
+// and returns the result. The handle is closed at the end.
+func ReadByteDataFromBus(ctx context.Context, bus I2C, addr, register byte) (byte, error) {
```

Comment:
[minor] I get that these are useful helper functions for the board, but since we've just cleaned up and pared down the interface, can you move this functionality to the am5 encoder side?